backend/app/services/knowledge/knowledge_service.py

```
@@ -481,13 +584,31 @@ def update_knowledge_base(
+                restricted_fields = [
+                    "retrieval_config",
+                    "summary_enabled",
+                    "summary_model_ref",
+                    "max_calls_per_conversation",
+                    "exempt_calls_before_check",
+                ]
+                for field in restricted_fields:
+                    if getattr(data, field, None) is not None:
+                        raise ValueError(
+                            "Developer can only update name and description"
+                        )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)